### ISSUE TITLE
Setting "selector-no-redundant-nesting-selector" to true causes FATAL error

### ISSUE THREAD

**Issue by stphane** [kristerkari/stylelint-scss#266]
Setting option `scss/selector-no-redundant-nesting-selector: true` in _.stylelintrc.js_ causes:
```
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
==== JS stack trace =========================================

Security context: 0x3dc4aaf25879 <JSObject>
    1: tokenize(aka tokenize) [/home/workspace/project/node_modules/stylelint-scss/node_modules/postcss-selector-parser/dist/tokenize.js:~86] [pc=0x121715afd89b](this=0x31d4e50822d1 <undefined>,input=0x390ce8826d9 <Object map = 0x68c2f87b361>)
    2: new Parser [/home/workspace/project/node_modules/stylelint-scss/node_modules/postcss-selector-parser/dist/parser.js:159] ...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x55fab5d742c2 [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [node]
 6: v8::internal::Runtime_AllocateInTargetSpace(int, v8::internal::Object**, v8::internal::Isolate*) [node]
 7: 0x1217159042fd
```
While running:  `webpack-dev-server --env.env dev --env.isLocal true --progress --inline --port=8080 --watch --hot`

Using
`└─┬ stylelint-scss@3.2.0`
`.        └── postcss-selector-parser@4.0.0`

**Comment by stphane** [comment]
Thank you for your feedback, that's annoying actually, I could hardly imagine how my setup would interfer with this switch option which barely seems to check legitimate use of `&` ref during parse process.  
This option used to work but now broke my build process since I have updated some dependencies, but sources have not changed.

**Comment by kristerkari** [comment]
@stphane If you upload your project somewhere where I can access it, I can have a look to see if I manage to reproduce the error.

**Comment by stphane** [comment]
Thank you for proposing but I cannot expose this project publicly `:|`

**Comment by kristerkari** [comment]
Alright. Just let me know if you find any reason why the error happens. For now I don't think that there is much that I can do to help. :)

**Comment by rtibbles** [comment]
I have come across this error too, but even with the setting in question turned off.
I think the error is actually coming from one of the dependencies, see here: https://github.com/postcss/postcss-selector-parser/issues/139

It has been fixed in the 0.5.0 release candidate, but there is no final release just yet.

**Comment by rtibbles** [comment]
From my investigations I narrowed it down to this rule: `'scss/percent-placeholder-pattern'` rather than the one reported above, with the following setting `"^[a-z]+([a-z0-9-]+[a-z0-9]+)?$"` in `stylelint-config-sass-guidelines` package.

**Comment by kristerkari** [comment]
@rtibbles Thanks a lot for investigating why the error happens. I'll try to see if I manage reproduce the error on my machine.

**Comment by rtibbles** [comment]
One other useful thing to note, is that I only triggered this when linting somewhere on the order of 200+ files (and maintaining reference to each file by wrapping all calls to stylelint in a `Promise.all`).

**Comment by kristerkari** [comment]
I think that when postcss-selector-parser v5.0.0 gets released, we can upgrade to it and see if that fixes this issue.

**Comment by bjankord** [comment]
@rtibbles, you might try making a branch/fork of this repo and update postcss-selector-parser to pull in the latest release candidate of [postcss-selector-parser, v5.0.0-rc.4](https://github.com/postcss/postcss-selector-parser/releases), and see if that resolves the issue. I haven't been able to reproduce the noted memory issue.

**Comment by rtibbles** [comment]
I just stopped using the rule I identified above, and that stopped me getting the memory issues, so I am not in an urgent rush to fix this!

**Comment by bjankord** [comment]
Ok, that is understandable. I don't have a reproducible test case, so likewise,  I am not in an urgent rush to fix this.

**Comment by kristerkari** [comment]
This should now be fixed in v3.4.4. If it's not fixed yet, let's re-open this issue.

https://github.com/kristerkari/stylelint-scss/releases/tag/3.4.4
